File: src/main/java/com/lmp/web/controller/api/v1/dto/AppointmentResponse.java
package com.lmp.web.controller.api.v1.dto;

import com.lmp.domain.entity.Appointment;

import java.time.LocalDateTime;
import java.util.UUID;

/**
 * DTO de réponse pour un rendez-vous.
 */
public record AppointmentResponse(
        UUID id,
        String clientName,
        String clientEmail,
        String clientPhone,
        LocalDateTime appointmentDate,
        String subject,
        String description,
        String status,
        Integer durationMinutes,
        Integer priority,
        LocalDateTime createdAt,
        LocalDateTime confirmedAt
) {
    public static AppointmentResponse from(Appointment appointment) {
        return new AppointmentResponse(
                appointment.getId(),
                appointment.getClientName(),
                appointment.getClientEmail(),
                appointment.getClientPhone(),
                appointment.getAppointmentDate(),
                appointment.getSubject(),
                appointment.getDescription(),
                appointment.getStatus() != null ? appointment.getStatus().name() : null,
                appointment.getDurationMinutes(),
                appointment.getPriority(),
                appointment.getCreatedAt(),
                appointment.getConfirmedAt());
    }
}

File: backend-dotnet/src/Lmp.Application/Crm/AppointmentResponse.cs
using Lmp.Domain.Crm;

namespace Lmp.Application.Crm;

/// <summary>Appointment response DTO. Faithful port of <c>com.lmp.crm.dto.AppointmentResponse</c>.</summary>
public sealed record AppointmentResponse(
    Guid Id,
    string? ClientName,
    string? ClientEmail,
    string? ClientPhone,
    DateTime AppointmentDate,
    string Subject,
    string? Description,
    string? Status,
    int? DurationMinutes,
    int? Priority,
    DateTime CreatedAt,
    DateTime? ConfirmedAt)
{
    public static AppointmentResponse From(Appointment a)
        => new(
            a.Id,
            a.ClientName,
            a.ClientEmail,
            a.ClientPhone,
            a.AppointmentDate,
            a.Subject,
            a.Description,
            a.Status.ToWire(),
            a.DurationMinutes,
            a.Priority,
            a.CreatedAt,
            a.ConfirmedAt);
}
